### PR TITLE
Integrated BLSQ/openimis-be-mobile_py

### PR DESCRIPTION
**Pull request by malinowskikam**
Integrated ``MobileEnrollmentMutation`` from https://github.com/BLSQ/openimis-be-mobile_py/

[https://openimis.atlassian.net/browse/OP-2038](https://openimis.atlassian.net/browse/OP-2038)

### CODE REVIEW

**Comment by dragos-dobre** [comment]
@malinowskikam Did you run the tests locally? Please fix based on test errors.

File: controls/apps.py [hunk 1]
```diff
@@ -1,5 +1,44 @@
...
+DEFAULT_CFG = {
+    "gql_mutation_create_families_perms": ['101002'],
+    "gql_mutation_update_families_perms": ['101003'],
+    "gql_mutation_create_insurees_perms": ["101102"],
+    "gql_mutation_update_insurees_perms": ["101103"],
+    "gql_mutation_create_policies_perms": ['101202'],
+    "gql_mutation_edit_policies_perms": ['101203'],
+    "gql_mutation_renew_policies_perms": ["101205"],
+    "gql_mutation_create_premiums_perms": ["101302"],
+    "gql_mutation_update_premiums_perms": ["101303"],
```
Comment (dragos-dobre) [inline controls/apps.py]: These are already defined in the specific modules and should not be defined/configured here. You should take the rights from the specific modules as they might be configured differently. Single point of truth.

File: controls/tests/test_models.py [hunk 2]
```diff
@@ -23,22 +20,20 @@ def test_basic_creation(self):
         self.assertEqual(control.usage, 'a_form')
         self.assertEqual(str(control), 'Field a_field (Optional) for forms a_form')
 
-    def test_invalid_adjustability(self):
-        invalid_adjustability_values = [
+    def test_unvalid_adjustability(self):
+        unvalid_adjustability_values = [
```
Comment (dragos-dobre) [inline controls/tests/test_models.py]: The right terminology is invalid and not unvalid. Please change this in all occurrences.

File: controls/gql_mutations.py [hunk 3]
```diff
@@ -0,0 +1,160 @@
...
+
+    @classmethod
+    def async_mutate(cls, user, **data):
+        logger.info("Receiving new mobile enrollment request")
+        logger.info(data)
+        try:
+            if type(user) is AnonymousUser or not user.id:
+                raise ValidationError("mutation.authentication_required")
+            if any(not user.has_perms(right) for right in MOBILE_ENROLLMENT_RIGHTS):
+                raise PermissionDenied("unauthorized")
```
Comment (dragos-dobre) [inline controls/gql_mutations.py:75]: I was wondering if we should let the specific services to check this. What happens if there are errors (e.g. existing insuree id) in the data with the already saved data ?

**Review by dragos-dobre — CHANGES_REQUESTED** [review]

File: controls/tests/test_models.py (hunk 2 above)
Comment (malinowskikam) [reply controls/tests/test_models.py]: Fixed

File: controls/apps.py (hunk 1 above)
Comment (malinowskikam) [reply controls/apps.py]: Fixed

File: controls/gql_mutations.py (hunk 3 above)
Comment (malinowskikam) [reply controls/gql_mutations.py:75]: I think validation is already handled by the service calls in lines 92 and 99. My guess is they added this here as an early return, canceling the operation early if the user does not have any of the required (similar to validate_mutation we have in newer modules)

**Review by dragos-dobre — APPROVED** [review]

**Pull request merged**